src/librustc_mir/interpret/memory.rs

```diff
@@ -83,19 +77,16 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
     pub fn allocate_value(
         &mut self,
         alloc: Allocation,
-        kind: Option<MemoryKind<M::MemoryKinds>>,
+        kind: MemoryKind<M::MemoryKinds>,
     ) -> EvalResult<'tcx, AllocId> {
         let id = self.tcx.alloc_map.lock().reserve();
         M::add_lock(self, id);
         match kind {
-            Some(kind @ MemoryKind::Stack) |
-            Some(kind @ MemoryKind::Machine(_)) => {
+            kind @ MemoryKind::Stack |
+            kind @ MemoryKind::Machine(_) => {
                 self.alloc_map.insert(id, alloc);
```

Comment:
I think these two maps can now be merged into one

Reply:
Oops, yep.